source/MaterialXRuntime/RtNodeGraph.cpp

```diff
@@ -285,4 +287,12 @@ string RtNodeGraph::asStringDot() const
     return dot;
 }
 
+RtTokenVec RtNodeGraph::getEditableMetadata() const
+{
+    RtTokenVec editableMetadata = RtNode::getEditableMetadata();
+    editableMetadata.push_back(RtToken(Element::FILE_PREFIX_ATTRIBUTE.c_str()));
+    editableMetadata.push_back(RtToken(Element::COLOR_SPACE_ATTRIBUTE.c_str()));
+    return editableMetadata;
```

Comment:
Since this is static information we don't need to create the list every time. You could keep the list as a local variable in an anonymous namespace in this file. And return this by const reference.

Reply:
Still not quite following this. So RtSchema will have a const RtTokenVec for it's public metadata, and so will RtNodeGraph. But then, how do I combine both of these lists and still return a const ref? Probably missing something obvious...

Reply:
Since most schemas has no inheritance hierarchy at all, and the exception like node/nodegraph only have one level deep hierarchy, I don't think it's worth gathering these names hierarchically. Instead we can just keep a separate constant  RtTokenVec per schema class, with all names that are defined for that schema.

This way we avoid the costly dynamic allocation and construction of these vectors for every call. It's something that is done all over the place in the core API, and I've been trying to avoid that when possible in the runtime API, to have better performance.

Reply:
But shouldn't a nodegraph inherit the metadata of a node? Not that I currently have any node metadata, but still.

Reply:
It doesn't have to be done that way just because the nodegraph class is derived from the node class. If it's much more efficient to store as separate lists in this case why not do so..

The fact that nodegraph is derived from node in this API is because it's convenient for other reasons. The MaterialX specification has no such inheritance specified between node and nodegraph. So keeping separate lists is totally fine, and efficient :)